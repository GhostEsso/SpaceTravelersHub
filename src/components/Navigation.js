import { NavLink } from 'react-router-dom';
import planet from '../assets/planet.png';
import styles from '../styles/navigation.module.css';

const Navigation = () => (
  <header className={styles.navbar}>
    <div className={styles.navbarContent}>
      <div className={styles.logoSection}>
        <img src={planet} alt="Space Travelers' Hub Logo" />
        <h1>Space Travelers&apos; Hub</h1>
      </div>

      <nav>
        <ul className={styles.navLinks}>
          <li>
            <NavLink
              to="/"
              className={({ isActive }) => `${styles.navLink} ${isActive ? styles.activeLink : ''}`}
            >
              HOME
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/rockets"
              className={({ isActive }) => `${styles.navLink} ${isActive ? styles.activeLink : ''}`}
            >
              ROCKETS
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/missions"
              className={({ isActive }) => `${styles.navLink} ${isActive ? styles.activeLink : ''}`}
            >
              MISSIONS
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/my-profile"
              className={({ isActive }) => `${styles.navLink} ${isActive ? styles.activeLink : ''}`}
            >
              PROFILE
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  </header>
);

export default Navigation;
