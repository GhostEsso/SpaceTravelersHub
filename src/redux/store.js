import { configureStore } from '@reduxjs/toolkit';
import Rocketslice from './rockets/RocketsSlice';
import MissionSice from './Mission/MissionSice';

const store = configureStore({
  reducer: {
    rockets: Rocketslice,
    mission: MissionSice,
  },
});

export default store;
