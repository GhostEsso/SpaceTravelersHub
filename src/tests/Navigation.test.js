import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { it, expect } from '@jest/globals';
import Navigation from '../components/Navigation';
import store from '../redux/store';

it('renders correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
