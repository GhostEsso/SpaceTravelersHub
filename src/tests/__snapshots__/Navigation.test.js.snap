// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="border-bottom mb-5 py-2"
  >
    <div>
      <img
        alt="logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/my-profile"
          >
            My profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
